hdfs/client.py

```diff
@@ -599,7 +602,7 @@ def wrap(_reader, _chunk_size, _progress):
         for path_tuple in fpath_tuples:
           _upload(path_tuple)
       else:
-        _map_async(n_threads, _upload, fpath_tuples)
+        _map_async(n_threads, _upload, fpath_tuples, timeout=timeout)
```

Comment:
It is confusing that this feature only works when there is more than one thread. We should keep the two behaviors consistent. It's also not idea that it won't work for Python 2.6, but that's OK given that it's not officially supported anymore.